### PR TITLE
adds long process support

### PR DESCRIPTION
**Pull request by bardiarastin**

### CODE REVIEW

File: README.md
Comment (jadjoubran) [inline README.md:6]: nice catch!  😄

File: README.md
Comment (jadjoubran) [inline README.md]: Can you please change the title to `Configuration`
just like in https://github.com/jadjoubran/webdash-performance-budget#configuration

End users will have this configuration in `webdash.json` already (we'll work on a fix for it on the main repo later)

I just realized that `longRunningScripts` means that they just take too long to finish
so how about we call them `serverScripts`?

File: README.md
Comment (jadjoubran) [inline README.md]: let's use `"dev", "start"` as examples, which are quite common

File: api.js
Comment (jadjoubran) [inline api.js]: why do longRunningScripts need to be stored in an object different than the one that already exists? which is `processes`

File: api.js
Comment (jadjoubran) [inline api.js]: if we're not gonna show the output in the terminal, is there a reason why you chose sync over exec?

**Review by jadjoubran — CHANGES_REQUESTED** [review]

**Comment by jadjoubran** [comment]
okay I did a first test & review
running a task did work (in the background) but clicking on the button again would re-run the same script (rather than stop it)

Why not just keep the loader next to it, to signal that it's running and then on hover show the stop button? (Btw this should be working already.. I'll upload a screencast/gif to show you => let me know if it doesn't work for you)
Let's make sure we figure this out before you fix the PR please just so that we don't waste your time
Thanks!

File: README.md
Comment (bardiarastin) [reply README.md]: agree with changing longRunningScripts to serverScripts 👍
for Configuration there already exists a title, i can remove Long running tasks or write configuration for sever scripts instead, what's your opinion ?

File: api.js
Comment (bardiarastin) [reply api.js]: i've used spawn because of https://stackoverflow.com/a/10232330/7091035

File: README.md
Comment (jadjoubran) [reply README.md]: yes please add it at the end of #Configuration

**Comment by bardiarastin** [comment]
reviews done in e90d92d (sorry for dummy commit message, i'm not that awesome in English 😄)
issues:
1:
killing processes is not safe (some processes with child processes inside don't get killed correctly) 

idea: we can use https://github.com/pkrumins/node-tree-kill

2: 
about keeping the loading everything is as you wanted but there are problems, right now it's working like: 
1: send start request
2: add loading icon
3: wait for response and then remove loader
but because in long running processes we never send a response loading stays till request timeout error

idea: we can send some object like { keepLoading: true, message: 'process dev started successfully' } or { serverScript: true, message: 'process dev started successfully' }

File: api.js
Comment (jadjoubran) [inline api.js]: Please use `req.app.locals.config` instead of this line, just like in [webdash-pwa-manifest api.js#8](https://github.com/jadjoubran/webdash-pwa-manifest/blob/master/api.js#L8)

File: api.js
Comment (jadjoubran) [inline api.js:3]: shouldn't this be reflected in package.json? added to dependencies

**Review by jadjoubran — CHANGES_REQUESTED** [review]

**Comment by jadjoubran** [comment]
okay so I tested it end to end locally, and:

1. yes please let's go with your recommendation keepLoading: true, but let's name it: serverStarted: true
2. after stopping, the loader keeps on running
3. if you need any the polymer side, feel free to ask me or you can just leave it to me and I'll work on it

Thanks!

**Comment by bardiarastin** [comment]
all things done
working like a charm
sorry for delay

File: api.js
Comment (jadjoubran) [inline api.js]: I think we should also pass in `--silent` just like I do on line 65 to skip all the typical npm logs that are not useful

File: webdash-npm-scripts.html
Comment (jadjoubran) [inline webdash-npm-scripts.html]: Shouldn't we keep sending the `, {script}` here?
I was able to reproduce a bug with this change:
If you have 2 server scripts that are running, and you stop the first one, then the 2nd one
only the first one will be stopped, because you're not sending the `script`

**Review by jadjoubran — COMMENTED** [review]
These are the last 2 changes <3
Thanks!

**Comment by jadjoubran** [comment]
After merging this, let me know if you'd like to send a PR for webdash (main repo) to add `serverScripts` to the `webdash.json` config (issue tracked [here](https://github.com/jadjoubran/webdash/issues/59))

**Comment by bardiarastin** [comment]
of course, why not 😍

File: webdash-npm-scripts.html
Comment (bardiarastin) [reply webdash-npm-scripts.html]: oh, this is a typo 😬

**Comment by bardiarastin** [comment]
all done in dc07b6a, ready to merge

**Comment by jadjoubran** [comment]
Thanks so much @bardiarastin!!! 😄 🎉 
This will be available in the next release (in the next few days)

**Comment by bardiarastin** [comment]
🎉🎉🎉 you can find me on twitter with same user name as github, i've followed you 😜

**Pull request merged**
